shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -261,9 +271,17 @@ - (void)applicationBecameActive:(NSNotification*)notification {
 }
 
 - (void)applicationWillResignActive:(NSNotification*)notification {
+  [_lifecycleChannel.get() sendMessage:@"AppLifecycleState.inactive"];
+}
+
+- (void)applicationDidEnterBackground:(NSNotification*)notification {
   [_lifecycleChannel.get() sendMessage:@"AppLifecycleState.paused"];
 }
 
+- (void)applicationWillEnterForeground:(NSNotification*)notification {
+  [_lifecycleChannel.get() sendMessage:@"AppLifecycleState.inactive"];
```

Comment:
Oh wait. I see. The "will enter foreground" is followed by the "did become active". That was confusing to me.